### PR TITLE
Picturae import

### PR DESCRIPTION
**Pull request by delarocamcas**
Files Created:
-remove_incomplete_attachments.py - as script for removing unattached images or incomplete attachment records on specify.
-PIC_database_updater.py - an updater script used for udating peace-meal pieces of data to collection records , so digileap and NfN can update data on a rolling basis. Some quality control measures to be added in a future PR.

Other smaller changes:
-sql_csv_tools. the create_update_statement function automatically changes the TimeStampModified field to be the most recent datetime upon update.
-logger_int no longer a parameter for methods in SqlCsvTools, instead made into an optional param that is inherited from the parent class upon init.
-erroneous character in R_RNRS.R removed.
-collection name added as config variable for picturae_config.

### CODE REVIEW

File: PIC_database_updater.py
Comment (delarocamcas) [inline PIC_database_updater.py:1]: draft of class for updating individual specify database fields, to be used as we get piece-meal data from notes from nature and digileap. Will be expanded to include quality control measures as needed.

File: image_client.py [hunk 1]
```diff
@@ -150,24 +150,12 @@ def check_image_db_if_filename_imported(self, collection, filename, exact=False)
             'search_type': 'filename',
             'token': self.generate_token(filename)
         }
-        return self.decode_response(params)
 
-    # works for full filepath and original extension. "exact" does a sql "like" operation
```
Comment (delarocamcas) [inline image_client.py:155]: this function was an identical duplicate of an existing function in image_client

File: monitoring_tools.py [hunk 2]
```diff
@@ -243,5 +245,9 @@ def send_monitoring_report(self, subject, time_stamp):
             msg['From'] = "ibss-crontab@calacademy.org"
             msg['To'] = email
             msg['Subject'] = subject
+            # with smtplib.SMTP(port=self.config.smtp_port, host=self.config.smtp_server) as server:
```
Comment (delarocamcas) [inline monitoring_tools.py:248]: commented out code for configuring using gmail

File: picturae_config.template.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import os
+COLLECTION_NAME = 'collectionname'
```
Comment (delarocamcas) [inline picturae_config.template.py:2]: added collection name to picturae_config

File: picturae_csv_create.py
Comment (delarocamcas) [inline picturae_csv_create.py:1]: all the changes in this file are just removing logger as an argument for sql_csv_tools methods

File: picturae_readme.txt [hunk 4]
```diff
@@ -53,7 +53,7 @@ file list:
                     is run under the purge function in client_tools.py example command using
                     MD5 batch code from error tracking db.
             example:
```
Comment (delarocamcas) [inline picturae_readme.txt:55]: updated example to reflect current directory in readme

File: specify7_ipup.py [hunk 5]
```diff
@@ -23,10 +24,9 @@ def set_direct():
             none
             """
     # get current directory
-    #current_directory = os.getcwd()
+    # current_directory = os.getcwd()
     # this is absolute path for running chron from bin
-    current_directory = "/Users/mdelaroca/Documents/sandbox_db/" \
-                        "specify-sandbox/specify-asset-importer"
+    current_directory = os.path.dirname(os.path.abspath(__file__))
```
Comment (delarocamcas) [inline specify7_ipup.py:29]: changed to get filepath directory regardless of who's machine it is operating on

File: sql_csv_utils.py [hunk 6]
```diff
@@ -196,25 +201,31 @@ def create_update_statement(self, tab_name: str, col_list: list, val_list: list,
                 val_list: list of values with which to update above list of columns(order matters)
                 condition: condition sql string used to select sub-sect of records to update.
         """
-        update_string = " SET"
+        update_string = f''' SET TimestampModified = "{time_utils.get_pst_time_now_string()}",'''
```
Comment (delarocamcas) [inline sql_csv_utils.py:204]: automatically updates timestamp modified when updating

File: temp_utility_scripts/remove_incomplete_attachments.py
Comment (delarocamcas) [inline temp_utility_scripts/remove_incomplete_attachments.py:1]: this class is used to remove images without attachment records and associated barcodes , as well as attachments with missing colleciton object attachments.

File: tests/sqlite_csv_utils.py
Comment (delarocamcas) [inline tests/sqlite_csv_utils.py:1]: added logger to sqlite derived class from SqlCsvTools

**Review by foozleface — APPROVED** [review]
looks good, and nice call on the order of checking and deleting from image server before doing the sql deletion in specify.

**Pull request merged**